admin/client/src/components/form-builder/index.js

```diff
@@ -119,6 +151,101 @@ export class FormBuilderComponent extends SilverStripeComponent {
   }
 
   /**
+   * Update handler passed down to each form field as a prop.
+   * Form fields call this method when their state changes.
+   *
+   * You can pass an optional callback as the third param. This can be used to
+   * implement custom behaviour. For example you can use `createFn` hook from
+   * your controller context like this.
+   *
+   * controller.js
+   * ...
+   * detailEditFormCreateFn(Component, props) {
```

Comment:
This requires access to the `<FormBuilder>` instantiation, which no code outside core will have (DI won't help there). I think we need a static registration system, similar to the `reducer-register` - but the code inside that callback would look similar to the example here.